### PR TITLE
KTOR-6123 Update kotlin to 1.9.0

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-6123](https://youtrack.jetbrains.com/issue/KTOR-6123/Update-Kotlin-to-1.9.0)

### CODE REVIEW

**Review by rsinukov — COMMENTED** [review]
Please check CI

**Comment by e5l** [comment]
Done, please take a look

**Comment by rsinukov** [comment]
native builds are still failing

File: ktor-io/posix/src/io/ktor/utils/io/bits/MemoryPrimitivesNative.kt [hunk 1]
```diff
@@ -1,10 +1,11 @@
-@file:Suppress("NOTHING_TO_INLINE", "ConstantConditionIf")
+@file:Suppress("OTHING_TO_INLINE", "ConstantConditionIf")
```
Comment (rsinukov) [inline ktor-io/posix/src/io/ktor/utils/io/bits/MemoryPrimitivesNative.kt:1]: typo?

**Review by rsinukov — APPROVED** [review]

**Pull request merged**